### PR TITLE
RenderLine fix when no options specified

### LINKED ISSUE #2044: Error when trying to render a line with 2 colors
Works fine when drawing a single line color. Error happens when trying to use 2 colors:

```
launch.js:7876 TypeError: Cannot read property 'layer' of undefined
    at Application._addLines (playcanvas-stable.dbg.js:48205)
    at Application.renderLine (playcanvas-stable.dbg.js:48273)
```

The example, [shown in the docs](https://developer.playcanvas.com/en/api/pc.Application.html#renderLine), will generate the error.
```
var start = new pc.Vec3(0, 0, 0);
var end = new pc.Vec3(1, 0, 0);
var startColor = new pc.Color(1, 1, 1);
var endColor = new pc.Color(1, 0, 0);
app.renderLine(start, end, startColor, endColor);
```

### PR DESCRIPTION
**Pull request by mvaligursky**
fixes #2044

### CODE REVIEW

File: src/scene/immediate.js [hunk 1]
```diff
@@ -266,8 +266,10 @@ Object.assign(pc.Application.prototype, function () {
         } else if (arg3) {
             // options passed in
             options = arg3;
-        } else {
-            // no arg3, use default options
+        }
+
+        // use default options
+        if (!options) {
```
Comment (willeastcott) [inline src/scene/immediate.js:272]: Would it be better to fix `_addLines` to handle an undefined options object? That way, you would avoid allocating a JS object on every call to this function.

Comment (mvaligursky) [reply src/scene/immediate.js]: done, good idea.

File: src/scene/immediate.js [hunk 2]
```diff
@@ -134,29 +134,29 @@ Object.assign(pc.Application.prototype, function () {
     }
 
     function _addLines(position, color, options) {
-        if (options.layer === undefined) options.layer = this.scene.layers.getLayerById(pc.LAYERID_IMMEDIATE);
-        if (options.depthTest === undefined) options.depthTest = true;
+        var layer = (options && options.layer) ? options.layer : this.scene.layers.getLayerById(pc.LAYERID_IMMEDIATE);
+        var depthTest = (options && options.depthTest) ? options.depthTest : true;
```
Comment (willeastcott) [inline src/scene/immediate.js:138]: What if `options.depthTest` is set to `false`?

File: src/scene/immediate.js [hunk 3]
```diff
@@ -134,29 +134,29 @@ Object.assign(pc.Application.prototype, function () {
     }
 
     function _addLines(position, color, options) {
-        if (options.layer === undefined) options.layer = this.scene.layers.getLayerById(pc.LAYERID_IMMEDIATE);
-        if (options.depthTest === undefined) options.depthTest = true;
+        var layer = (options && options.layer) ? options.layer : this.scene.layers.getLayerById(pc.LAYERID_IMMEDIATE);
```
Comment (willeastcott) [inline src/scene/immediate.js:137]: What if I pass `pc.LAYERID_WORLD` which is set to 0?  
https://github.com/playcanvas/engine/blob/master/src/scene/scene.js#L150

File: src/scene/immediate.js [hunk 4]
```diff
@@ -134,29 +134,29 @@ Object.assign(pc.Application.prototype, function () {
...
+        var depthTest = (options && options.depthTest !== undefined) ? options.depthTest : true;
+        var mask = (options && options.mask) ? options.mask : undefined;
```
Comment (willeastcott) [inline src/scene/immediate.js:139]: This one's a bit trickier. We don't document `pc.MeshInstance#mask` very well. There are some enums here:
https://github.com/playcanvas/engine/blob/master/src/scene/scene.js#L576
This line will convert 0 to undefined. Either way, the `if` statement below will fail. So I think this is fine.

File: src/scene/immediate.js (hunk 3 above)
Comment (mvaligursky) [reply src/scene/immediate.js:137]: that should be ok .. you pass in Layer object, not id. That's how this is called:

     * var worldLayer = app.scene.layers.getLayerById(pc.LAYERID_WORLD);
     * app.renderLine(start, end, color, {
     *     layer: worldLayer
     * });

File: src/scene/immediate.js (hunk 4 above)
Comment (mvaligursky) [reply src/scene/immediate.js:139]: yep that's what I think as well, this should be fine.

File: src/scene/immediate.js (hunk 3 above)
Comment (willeastcott) [reply src/scene/immediate.js:137]: Oh yes, you're right. 👍

**Pull request merged**